### ISSUE TITLE
batchnorm2d  track_running_stats

### ISSUE THREAD

**Issue by luhc15** [pytorch/pytorch#8430]
when I load a pretrained model, Runtime error occurred:  Missing key(s) in state_dict: bn.num_batches_tracked in every BatchNorm2d layer

**Comment by pfriesch** [comment]
I have the same problem, trained the model with pytorch 0.4.0 and tried to load it with 0.4.1.

**Comment by Ai-is-light** [comment]
I have the same problem, trained the model with pytorch 0.3.1 and tried to load it with 0.4.0.

**Comment by SsnL** [comment]
@pfriesch @Ai-is-light @LukasMosser Can you provide your model definition, `list(state_dict.keys())`, and `getattr(state_dict, _metadata, None)` please? Thanks!

**Comment by mojesty** [comment]
Hello! I have the same issue when trying to load in `0.4.1.post2` model trained in _mayve_ `0.3` (authors of the code didn't write requirements file so I have to infer version myself, but it is not 0.3 + because code still have Variables and volatiles).
In my case, `getattr(model.state_dict(), '_metadata', None)` returned 
```
OrderedDict([('', {'version': 1}),
             ('drop', {'version': 1}),
             ('idrop', {'version': 1}),
             ('rdrop', {'version': 1}),
             ('encoder', {'version': 1}),
             ('parser', {'version': 1}),
             ('parser.drop', {'version': 1}),
             ('parser.gate', {'version': 1}),
             ('parser.gate.0', {'version': 1}),
             ('parser.gate.1', {'version': 1}),
             ('parser.gate.2', {'version': 2}),
             ('parser.gate.3', {'version': 1}),
             ('parser.gate.4', {'version': 1}),
             ('parser.gate.5', {'version': 1}),
             ('parser.gate.6', {'version': 1}),
             ('reader', {'version': 1}),
             ('reader.0', {'version': 1}),
             ('reader.0.drop', {'version': 1}),
             ('reader.0.memory_rnn', {'version': 1}),
             ('reader.0.memory_rnn.ih', {'version': 1}),
             ('reader.0.memory_rnn.ih.0', {'version': 1}),
             ('reader.0.memory_rnn.ih.1', {'version': 1}),
             ('reader.0.memory_rnn.hh', {'version': 1}),
             ('reader.0.memory_rnn.hh.0', {'version': 1}),
             ('reader.0.memory_rnn.hh.1', {'version': 1}),
             ('reader.0.memory_rnn.c_norm', {'version': 1}),
             ('reader.0.memory_rnn.drop', {'version': 1}),
             ('reader.0.projector_summ', {'version': 1}),
             ('reader.0.projector_summ.0', {'version': 1}),
             ('reader.0.projector_summ.1', {'version': 1}),
             ('reader.0.projector_summ.2', {'version': 1}),
             ('reader.1', {'version': 1}),
             ('reader.1.drop', {'version': 1}),
             ('reader.1.memory_rnn', {'version': 1}),
             ('reader.1.memory_rnn.ih', {'version': 1}),
             ('reader.1.memory_rnn.ih.0', {'version': 1}),
             ('reader.1.memory_rnn.ih.1', {'version': 1}),
             ('reader.1.memory_rnn.hh', {'version': 1}),
             ('reader.1.memory_rnn.hh.0', {'version': 1}),
             ('reader.1.memory_rnn.hh.1', {'version': 1}),
             ('reader.1.memory_rnn.c_norm', {'version': 1}),
             ('reader.1.memory_rnn.drop', {'version': 1}),
             ('reader.1.projector_summ', {'version': 1}),
             ('reader.1.projector_summ.0', {'version': 1}),
             ('reader.1.projector_summ.1', {'version': 1}),
             ('reader.1.projector_summ.2', {'version': 1}),
             ('predictor', {'version': 1}),
             ('predictor.drop', {'version': 1}),
             ('predictor.projector_pred', {'version': 1}),
             ('predictor.projector_pred.0', {'version': 1}),
             ('predictor.projector_pred.1', {'version': 1}),
             ('predictor.projector_pred.2', {'version': 1}),
             ('predictor.ffd', {'version': 1}),
             ('predictor.ffd.0', {'version': 1}),
             ('predictor.ffd.1', {'version': 1}),
             ('predictor.ffd.2', {'version': 2}),
             ('predictor.ffd.3', {'version': 1}),
             ('decoder', {'version': 1})])
```
and `list(model.state_dict().keys())` returned 
```['encoder.weight',
 'parser.gate.1.weight',
 'parser.gate.1.bias',
 'parser.gate.2.weight',
 'parser.gate.2.bias',
 'parser.gate.2.running_mean',
 'parser.gate.2.running_var',
 'parser.gate.5.weight',
 'parser.gate.5.bias',
 'reader.0.memory_rnn.ih.0.weight',
 'reader.0.memory_rnn.ih.0.bias',
 'reader.0.memory_rnn.ih.1.gamma',
 'reader.0.memory_rnn.ih.1.beta',
 'reader.0.memory_rnn.hh.0.weight',
 'reader.0.memory_rnn.hh.0.bias',
 'reader.0.memory_rnn.hh.1.gamma',
 'reader.0.memory_rnn.hh.1.beta',
 'reader.0.memory_rnn.c_norm.gamma',
 'reader.0.memory_rnn.c_norm.beta',
 'reader.0.projector_summ.1.weight',
 'reader.0.projector_summ.1.bias',
 'reader.1.memory_rnn.ih.0.weight',
 'reader.1.memory_rnn.ih.0.bias',
 'reader.1.memory_rnn.ih.1.gamma',
 'reader.1.memory_rnn.ih.1.beta',
 'reader.1.memory_rnn.hh.0.weight',
 'reader.1.memory_rnn.hh.0.bias',
 'reader.1.memory_rnn.hh.1.gamma',
 'reader.1.memory_rnn.hh.1.beta',
 'reader.1.memory_rnn.c_norm.gamma',
 'reader.1.memory_rnn.c_norm.beta',
 'reader.1.projector_summ.1.weight',
 'reader.1.projector_summ.1.bias',
 'predictor.projector_pred.1.weight',
 'predictor.projector_pred.1.bias',
 'predictor.ffd.1.weight',
 'predictor.ffd.1.bias',
 'predictor.ffd.2.weight',
 'predictor.ffd.2.bias',
 'predictor.ffd.2.running_mean',
 'predictor.ffd.2.running_var',
 'decoder.weight',
 'decoder.bias']
```
maybe there is some work around to set this attribute in newer version of pyotrch without changing source code?

**Comment by SsnL** [comment]
Thanks @mojesty . However, what I need is the results on the state_dict you are trying to load. Not the one of the module you are loading onto.

**Comment by SsnL** [comment]
@mojesty I'm glad to know that you found a workaround. Yes, switching to an old version works. But in the new version there is specific BC code to support load old state dict.. So could you please post those two results so I can know what is wrong? 0.4+ versions have much simpler API and many bug fixes so I want people to be able to use 0.4+ at lease.
